assets/app/data/services/upgrade/openstack-services.factory.js

```diff
@@ -0,0 +1,48 @@
+(function() {
+
+    angular
+        .module('crowbarData.upgrade')
+        .factory('openStackFactory', openStackFactory);
+
+    openStackFactory.$inject = ['$q', '$http'];
+    /* @ngInject */
+    function openStackFactory($q, $http) {
+        var factory = {
+            getOpenStackServices: getOpenStackServices,
+            stopOpenstackServices: stopOpenstackServices,
+            createOpenstackBackup: createOpenstackBackup
+        };
+
+        return factory;
+
+        function getOpenStackServices() {
+
+            var requestOptions = {
+                method: 'GET',
+                url: '/api/upgrade7/openstack-services'
```

Comment:
If changes on `app.js` are addressed, then you could store `/api/upgrade7/openstack-services` in a constant and reuse it on lines 32 and 42 to build the urls